pkg/kubelego/watch.go

```diff
@@ -88,13 +151,24 @@ func (kl *KubeLego) WatchEvents() {
 			oldIng.ResourceVersion = ""
 			upIng.ResourceVersion = ""
 
-			if !reflect.DeepEqual(oldIng, upIng) {
+			// we requeue ingresses only when their spec has changed, as the indicates
+			// a user has updated the specification of their ingress and as such we should
+			// re-trigger a validation if required.
+			if !reflect.DeepEqual(oldIng.Spec, upIng.Spec) {
```

Comment:
Are annotations  / labels on the ingress object examined by Kube-lego? Should we compare those also?

Reply:
Good point - if a user switches the `tls-acme` annotation from `false` to `true`, we won't notice it right now.